### DISCUSSION TITLE
flux did not work with calico network policies. flux did not know calico apis

### DISCUSSION THREAD

**Discussion by jipara** [fluxcd/flux2#2199] [Q&A]
hi I tried to deploy Global Network Policy and other calico network policy but Flux  cannot match for kind "GlobalNetworkPolicy" in version "projectcalico.org/v3"
I tried to deploy latest calico version 3.21.1 and I also tried 3.19.1 and I use EKS cluster version 1.19 and I also tried 1.21, flux v2 

![image](https://user-images.githubusercontent.com/50920745/145425258-7e3fe819-211f-4c9a-a25c-b89eeb8faf95.png)
Please let me know if Flux v2 is did not support calico network policies, so  I cannot find information in the internet.
I tried to email to calico team but they did not respond to me.

This is just example
apiVersion: projectcalico.org/v3
kind: GlobalNetworkPolicy
metadata:
  name: default-deny
spec:
  selector: projectcalico.org/namespace != "kube-system"
  types:
  - Ingress
  - Egress

**Comment by kingdonb** [comment]
This error `dry-run failed, error: no matches for kind "GlobalNetworkPolicy"` indicates that the cluster simply does not have GlobalNetworkPolicy CRD installed yet. Please make sure it's included in your manifests.

If you are using a HelmRelease to install Calico in the same breath as you install your CRs for network policy, that will fail, because the CRD hasn't been reconciled yet. I could try to guide you through using `dependsOn` for this, but since Kustomize Controller now has native support for reconciling CRDs at the same time as instances of the CR (it does the server-side apply in stages to facilitate this) it might be easier if you just don't use the Calico Helm chart, and opt for simple manifests instead? It looks like Calico offers both options.

Are you using Helm, does any of that help? :)
